STYLEGUIDE.md

```diff
@@ -0,0 +1,256 @@
+
+# Typedefs styleguide
+
+This styleguide aims to answer questions relevants to the cosmetics related to
+writing Idris programs.
+
+Since idris is a relatively young language, there is no general convention
+about its style. This document should serve as reference for current and future
+users of the typedefs codebase.
+
+## Line length: 120
+
+The maximum line length is 120 characters. In today's wide screen monitors
+there is no reason to limit ourselves to 80 characters, yet, having a maximum
+limit helps avoiding degenerate situations.
+
+## Indent style: 2 spaces
+
+Idris makes uses of some whitespace sensitive construction (typically the `do`
+notation), and also inherits from a culture of ascii-art code from Agda and
+Haskell. In order to allow this level of visual customisation and ensure
+consistency across editors, spaces are used.
+
+The language does not benefit from more than 2 space of indentation in most
+cases. Thought special constructions lend themselve to more spaces for
+indentation.
+
+### Indentation: where clauses
+
+``` idris
+fn : String
+fn = g
+  where
+    g : String
+    g = "hello"
+```
+
+Where clauses are aligned with 2 spaces
+
+### Indentation: Let declarations
+
+```idris
+fn : String
+fn = let greeting = "hello" in
+       greeting
+```
+
+Let declarations are indented with 2 spaces.
+
+Additional declarations are indented with 4 spaces while the expression is indented with 2 spaces
+
+```idris
+fn : String
+fn = let greeting = "hello"
+         whom = "world" in
+       greeting
+```
```

Comment:
this doesn't look that nice, from my point of view, `greeting` being not aligned with everything. I'm afraid it won't be that readable in the middle of some complex code: Personally I second the @fredrikNordvallForsberg proposal

Reply:
I think I've come around to you and @fredrikNordvallForsberg but I'm still not convinced about the two spaces between `in` and the expression. What do you think of

```idris
let foo = 1
    bar = 2
 in expr
```

?

Reply:
looks nice, too!